aea/protocols/dialogue/base.py

```diff
@@ -857,30 +858,74 @@ def _basic_validation_non_initial_message(
                 ),
             )
 
-        last_message_id = self.last_message.message_id  # type: ignore
-        if message_id != last_message_id + 1:
-            return (
-                False,
-                "Invalid message_id. Expected {}. Found {}.".format(
-                    last_message_id + 1, message_id
-                ),
-            )
```

Comment:
We should have a simple method which checks correct incrementation.

Reply:
See above, a valid message id increments the max of all seen message ids by 1

Reply:
Message id must be evaluated as a tuple (message_id, sender), because only then do we have uniqueness.